public/views/partials/lib/commands/user/create.liquid

```diff
@@ -11,34 +11,43 @@
 {% liquid
   function object = 'modules/user/lib/commands/user/create/build', email: email, password: password, hook_params: hook_params
   function object = 'modules/user/lib/commands/user/create/check', object: object
+  function results = 'modules/core/lib/commands/hook/fire', hook: 'user_create_validate', params: hook_params
 
-  if object.valid
+  # Merge the array of hook results into a single object.
+  # Do all validation before taking any action.
```

Comment:
This block transforms the hook result into a single object that can be interpreted by the form template to field errors.

For example this hook result
```json
[
  {
    "errors": {
      "user_id": [
        "cannot be blank"
      ],
      "slug": [
        "cannot be blank"
      ],
      "date_of_birth": [
        "cannot be blank"
      ],
      "graduation_year": [
        "cannot be blank"
      ],
      "terms_and_conditions": [
        "Please accept the terms and conditions"
      ],
      "code_of_conduct": [
        "Please accept the Code of Conduct"
      ]
    },
    "valid": false
  },
{
    "errors": {
      "user_id": [
        "another issue"
      ],
      "slug": [
        "another issue"
      ],
      "date_of_birth": [
        "another issue"
      ],
      "graduation_year": [
        "another issue"
      ],
      "terms_and_conditions": [
        "another issue"
      ],
      "code_of_conduct": [
        "another issue"
      ]
    },
    "valid": false
  }
]
```

is transformed into
```json
{
  "errors": {
    "user_id": [
      "cannot be blank",
      "another issue"
    ],
    "slug": [
      "cannot be blank",
      "another issue"
    ],
    "date_of_birth": [
      "cannot be blank",
      "another issue"
    ],
    "graduation_year": [
      "cannot be blank",
      "another issue"
    ],
    "terms_and_conditions": [
      "Please accept the terms and conditions",
      "another issue"
    ],
    "code_of_conduct": [
      "Please accept the Code of Conduct",
      "another issue"
    ]
  },
  "valid": false
}
```